packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -144,10 +145,19 @@ export default createRule<[], MessageId>({
                 ]),
               ];
 
-              const stringValue = getStaticStringValue(expression);
-
-              if (stringValue != null) {
-                const escapedValue = stringValue.replace(/([`$\\])/g, '\\$1');
+              if (isLiteral(expression)) {
+                const escapedValue =
+                  typeof expression.value === 'string'
+                    ? // '1'   -> 1
+                      // '`'   -> \`
+                      // '${}' -> \${}
+                      // '\\'  -> \\
+                      expression.raw.slice(1, -1).replace(/([`$])/g, '\\$1')
+                    : // 1     -> 1
+                      // /`/   -> /\`/
+                      // /${}/ -> /\${}/
+                      // /\\/  -> /\\\\/
+                      String(expression.value).replace(/([`$\\])/g, '\\$1');
```

Comment:
Same here. I also wonder if we shouldn't do the template literal syntax escaping separately from this ternary to avoid some duplication.